src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -1013,7 +1026,15 @@ public bool StartsWith(string value, bool ignoreCase, CultureInfo? culture)
             return referenceCulture.CompareInfo.IsPrefix(this, value, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
         }
 
-        public bool StartsWith(char value) => Length != 0 && _firstChar == value;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool StartsWith(char value)
+        {
+            if (RuntimeHelpers.IsKnownConstant(value))
```

Comment:
```suggestion
            if (RuntimeHelpers.IsKnownConstant(value) && value != '\0')
```
Otherwise `string.Empty.StartsWith('\0')` would be wrong, right?

Reply:
Oops, right!

Reply:
Thanks!